examples/shader/custom_render_phase.rs

```diff
@@ -0,0 +1,630 @@
+//! This example demonstrates how to write a custom phase
+//!
+//! Render phases in bevy are used whenever you need to draw a group of meshes in a specific way.
+//! For example, bevy's main pass has an opaque phase, a transparent phase for both 2d and 3d.
+//! Sometimes, you may want to only draw a subset of meshes before or after the builtin phase. In
+//! those situations you need to write your own phase.
+//!
+//! This example showcases how writing a custom phase to draw a stencil of a bevy mesh could look
```

Comment:
I tried to google it but could only find some vague definitions; what is a 'stencil' in this context?

Reply:
It refers to essentially drawing the shape of the mesh. It's a bit of graphics programming terminology, but it's useful when you want to mask out something or want to create outlines. It's not really the purpose of the example though. It was just to have something a bit more concrete since it's most likely the most common use case for a custom phase.